neptune-core/src/application/util_proof/sent/mod.rs

```diff
@@ -0,0 +1,329 @@
+    ));
+    let lib_bagpeaks = library.import(Box::new(tasm_lib::mmr::bag_peaks::BagPeaks));
+    let release_date = library.kmalloc(1);
+    let lib_add_all_amounts_and_check_time_lock = library.import(
+        Box::new(crate::protocol::consensus::type_scripts::amount::add_all_amounts_and_check_time_lock::AddAllAmountsAndCheckTimeLock {
+            digest_source: crate::protocol::consensus::type_scripts::amount::total_amount_main_loop::DigestSource::
+            Hardcode(crate::protocol::consensus::type_scripts::native_currency::NativeCurrency.hash()),
+            release_date,
+    }));
+
+    // let rustfield_membershipproof = rustfield!(WitnessMemory::membership_proof);
+    let rustfield_senderrandomness = rustfield!(MsMembershipProof::sender_randomness);
+    let rustfield_utxo = rustfield!(WitnessMemory::utxo);
+    let rustfield_utxodigest = rustfield!(WitnessMemory::utxo_digest);
+    let rustfield_aoclleafindex = rustfield!(MsMembershipProof::aocl_leaf_index);
+
+    let payload = triton_asm! {
+        /* pasted from <../reserves>
+        ============== */
+        // _
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        {&rustfield_utxo}
+        {&Utxo::get_field("lock_script_hash")}
+        read_mem 5
+        pop 1
+        // [lock_script_digest]
+        write_io 5
+        // _
+
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        // *w
+        {&WitnessMemory::get_field("aocl")}
+        // *aocl
+
+        // prepare the value for '7.'
+        dup 0
+        {&MmrAccumulator::get_field("peaks")}
+        // *aocl *aocl_peaks
+
+        // read `receiver_digest`
+        read_io 5
+        // *aocl *aocl_peaks [receiver_digest]
+
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        // _ *w
+        {&rustfield_senderrandomness} // omits `{&rustfield_membershipproof}` from <../reserves>
+        read_mem {Digest::LEN}
+        pop 1
+        // *aocl *aocl_peaks [receiver_digest] [sender_randomness]
+
+        /* ## this segment diverges from the pasted code
+        *it only affects the output sequence* */
+        push 0
+        push 0
+        push 0
+        push 0
+        push 0
+        dup 9
+        dup 9
+        dup 9
+        dup 9
+        dup 9
+        hash
+        write_io 5
+
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        // _ *w
+        {&rustfield_utxo}
+        // *aocl *aocl_peaks [receiver_digest] [sender_randomness] *utxo
+    };
+
+    let main = triton_asm! {
+        read_mem 1
+        addi 2
+        // _ utxo_size *utxo+1
+        swap 1
+        // _ *utxo+1 utxo_size
+        call {lib_hash_varlen}
+        hint utxo_hash = stack[0..5]
+        // *aocl *aocl_peaks [receiver_digest] [sender_randomness] [utxo_hash]
+
+        push {FIRST_NON_DETERMINISTICALLY_INITIALIZED_MEMORY_ADDRESS}
+        {&rustfield_utxodigest}
+        read_mem 5
+        assert_vector
```

Comment:
Isn't the memory pointer still on top of the stack? If so, that will interfere with the `assert_vector`.